src/Aspire.Hosting.Azure/AzureResourcePreparer.cs

```diff
@@ -182,7 +182,11 @@ private async Task BuildRoleAssignmentAnnotations(DistributedApplicationModel ap
                         // attach the identity resource to compute resource so it can be used by the compute environment
                         resource.Annotations.Add(new AppIdentityAnnotation(identityResource));
```

Comment:
Should these lines be inside the `if` as well? We have an annotation that references itself here.

Reply:
Good call -- I don't think this has any impact since the `AppIdentityAnnotation` only gets queried for in the environment resources but better to avoid it altogether.